src/errors.rs

```diff
@@ -94,3 +109,50 @@ impl<T> From<PoisonError<T>> for ConfigurationAccessError {
         ConfigurationAccessError::LockAcquisitionError
     }
 }
+
+#[derive(Debug, Error)]
+pub(crate) enum SegmentEvaluationError {
+    #[error(transparent)]
+    SegmentEvaluationFailed(#[from] SegmentEvaluationErrorKind),
```

Comment:
I think we can declare this as follows, and remove one _indirection_

```suggestion
    #[error("Operation '{}' '{}' '{}' failed to evaluate: {}", segment_rule.attribute_name, segment_rule.operator, value, source)]
    SegmentEvaluationFailed {
        pub(crate) segment: Segment,
        pub(crate) segment_rule: SegmentRule,
        pub(crate) value: String,
        pub(crate) source: CheckOperatorErrorDetail,
    },
```

Reply:
We tried this yesterday, but it did not work, as we are not allowed to add visibility qualifiers into enum declaration :-(